pkg/resource/feature_group/testdata/feature_group/cr/fg_cmd_invalid_create_attempted.yaml

```diff
@@ -0,0 +1,25 @@
+apiVersion: sagemaker.services.k8s.aws/v1alpha1
+kind: FeatureGroup
+metadata:
+  creationTimestamp: null
+  name: unit-testing-feature-group
+spec:
+  eventTimeFeatureName: EventTime
+  featureDefinitions:
+  - featureName: TransactionID
+    featureType: Integral
+  - featureName: EventTime
+    featureType: Fractional
+  featureGroupName: ""
```

Comment:
why empty?

Reply:
After attempting to create a resource with an invalid feature group name, the resource that is created has that field as empty, along with setting the terminal status to true.

Reply:
that's strange, in case of error, sdkCreate returns `nil(latest), err` and so [the desired resource is passed to onError](https://github.com/aws-controllers-k8s/sagemaker-controller/blob/main/pkg/resource/feature_group/manager.go#L111-L113). [onError](https://github.com/aws-controllers-k8s/sagemaker-controller/blob/871401b98cdf1eacd57486ed1464b8a49730bba3/pkg/resource/feature_group/manager.go#L197) is just updating conditions so it should return the same cr with the terminal condition set to True

Reply:
If a terminal condition is set, which it is in the invalid create scenario, the onError function takes [this return statement](https://github.com/aws-controllers-k8s/sagemaker-controller/blob/871401b98cdf1eacd57486ed1464b8a49730bba3/pkg/resource/feature_group/manager.go#L210), which returns the resource with the terminal condition that was added from the sagemaker json response, accompanied by a separate error: ackerr.Terminal.